yb-voyager/src/cp/ybm/ybm.go

```diff
@@ -0,0 +1,727 @@
+			log.Infof("YBM API returned max sequence: %d for phase %d", maxSeq, phase)
+			return maxSeq, nil
+		}
+
+		// Check if it's a 404 (no prior data)
+		if strings.Contains(err.Error(), "404") {
+			log.Infof("GET migrations endpoint returned 404, using sequence 0 (first run or no data)")
+			return 0, nil
+		}
+
+		lastErr = err
+
+		// Check if error is retryable
+		if !isRetryableError(err) {
+			log.Warnf("Non-retryable error getting max sequence from YBM: %v. Falling back to sequence 0", err)
+			return 0, nil
+		}
+
+		log.Warnf("Attempt %d/%d failed to get max sequence from YBM: %v", attempt, maxAttempts, err)
+
+		if attempt < maxAttempts {
+			// Exponential backoff: 1s, 2s, 4s, 8s
+			backoff := time.Duration(1<<uint(attempt-1)) * time.Second
+			time.Sleep(backoff)
+		}
+	}
+
+	// After all retries failed, log warning and fallback to sequence 0
+	log.Warnf("Failed to get max sequence from YBM after %d attempts: %v. Falling back to sequence 0", maxAttempts, lastErr)
+	return 0, nil
+}
+
+// sendMigrationEvent sends a migration event to YBM API with retries
+func (ybm *YBM) sendMigrationEvent(event MigrationEvent) error {
+	urlPath := fmt.Sprintf("%s/api/public/v1/accounts/%s/projects/%s/clusters/%s/voyager/metadata",
+		ybm.config.Domain, ybm.config.AccountID, ybm.config.ProjectID, ybm.config.ClusterID)
+
+	var maxAttempts = 5
+	var lastErr error
+
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		err := ybm.postJSON(urlPath, event)
+		if err == nil {
+			log.Infof("Successfully sent migration event to YBM: phase=%d, sequence=%d, status=%s",
+				event.MigrationPhase, event.InvocationSequence, event.Status)
+			return nil
+		}
+
+		lastErr = err
+
+		// Check if error is retryable
+		if !isRetryableError(err) {
+			log.Warnf("Non-retryable error sending migration event to YBM: %v", err)
+			return err
+		}
+
+		log.Warnf("Attempt %d/%d failed to send migration event to YBM: %v", attempt, maxAttempts, err)
+
+		if attempt < maxAttempts {
+			// Exponential backoff: 1s, 2s, 4s, 8s
+			backoff := time.Duration(1<<uint(attempt-1)) * time.Second
+			time.Sleep(backoff)
+			// Update timestamp for retry
+			event.InvocationTimestamp = time.Now().Format(time.RFC3339)
+		}
+	}
+
+	return fmt.Errorf("failed to send migration event after %d attempts: %w", maxAttempts, lastErr)
+}
+
+// sendTableMetrics sends table metrics to YBM API with retries
+func (ybm *YBM) sendTableMetrics(metricsList []controlPlane.TableMetrics) error {
+	urlPath := fmt.Sprintf("%s/api/public/v1/accounts/%s/projects/%s/clusters/%s/voyager/table-metrics",
+		ybm.config.Domain, ybm.config.AccountID, ybm.config.ProjectID, ybm.config.ClusterID)
+
+	var maxAttempts = 5
+	var lastErr error
+
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		// Send each metric individually (YBM API expects single metric per PUT)
+		allSuccess := true
+		for _, metrics := range metricsList {
+			err := ybm.putJSON(urlPath, metrics)
```

Comment:
shouldn't you have retries at the level of each individual API rather than for the entire set?